conan/tools/microsoft/msbuilddeps.py

```diff
@@ -65,20 +65,54 @@ class MSBuildDeps(Generator):
         </Project>
         """)
 
+    def __init__(self, conanfile):
+        super(MSBuildDeps, self).__init__(conanfile)
+        self.configuration = conanfile.settings.build_type
+        self.platform = {'x86': 'Win32',
+                         'x86_64': 'x64'}.get(str(conanfile.settings.arch))
+        # TODO: this is ugly, improve this
+        self.output_path = os.getcwd()
+        # User configurable things
+        self._config_filename = None
+
+    def generate(self):
+        # TODO: Apply config from command line, something like
+        # configuration = self.conanfile.config.generators["msbuild"].configuration
+        # if configuration is not None:
+        #     self.configuration = configuration
+        # platform
+        # config_filename
+        # TODO: This is duplicated in write_generators() function, would need to be moved
+        # to generators and called from there
+        generator_files = self.content
+        for generator_file, content in generator_files.items():
+            generator_file_path = os.path.join(self.output_path, generator_file)
+            self.conanfile.output.info("Generating {}".format(generator_file))
+            save(generator_file_path, content)
+
     @property
     def filename(self):
         return None
 
-    @ staticmethod
-    def _name_condition(settings):
-        props = [("Configuration", settings.build_type),
-                 # FIXME: This probably requires mapping ARM architectures
-                 ("Platform", {'x86': 'Win32',
-                               'x86_64': 'x64'}.get(settings.get_safe("arch")))]
-
+    @property
+    def config_filename(self):
+        if self._config_filename is not None:
+            return self._config_filename
+        # Default name
+        props = [("Configuration", self.configuration),
+                 ("Platform", self.platform)]
         name = "".join("_%s" % v for _, v in props if v is not None)
+        return name.lower()
+
+    @config_filename.setter
+    def config_filename(self, value):
+        self._config_filename = value
```

Comment:
I'd remove this `config_filename` attribute, the user can already modify the name of the generated filename using `self.configuration` and `self.platform`, which are the arguments meaningful for VS IDE (`msbuild`) and are the ones considered in the __condition_.

Reply:
This was implemented because of your suggestions when thinking about this feature.
There is a use case, in which users can add explicitly specific generated .props files to their project files, without using the ``conan_depname.props`` that contains the condition.

I agree this seems a bit early, so I am going to remove it.